### DISCUSSION TITLE
서버 환경 세팅

### DISCUSSION THREAD

**Discussion by 2Ho0** [DKU-SWExpert-K8s/K8s#1] [질문]
환경 세팅 부분에서 패스워드 없이 ssh 접속이 가능하도록 설정하는 부분에 질문이 있습니다
각 서버마다 
`ssh- keygen`을 한 다음에
각 서버마다
`ssh-copy-id {Server Name}`
을 하면 되는건가요?

한 서버에 `ssh-copy-id {Server Name}`을 진행하면 현재 서버는 되는데 다른 서버는 time out이 뜨고
각 서버마다 진행하면 time out은 안뜨지만 `ssh {이동할 서버}`를 할 경우 time out이 뜹니다... 혹시 이 부분 해결하신 분 계신가요?

**Comment by cueon** [comment]
각 서버의 /etc/hosts 에
ubun20-01, ubun20-02, ubun20-03 에 대한 ip, hostname 을 적어주시고

각 서버에서 ssh-keygen 을 통해 개인키, 공개키를 생성한 이후에

ubun20-01 서버에서
ssh-copy-id ubun20-02
ssh-copy-id ubun20-03

ubun20-02 서버에서
ssh-copy-id ubun20-01
ssh-copy-id ubun20-03

ubun20-03 서버에서
ssh-copy-id ubun20-01
ssh-copy-id ubun20-02

해주시면 1번, 2번, 3번 서버 모두에서 서로 각각 패스워드 없이 접속 가능한 거로 알아요!

**Reply by cueon** [reply]
위 설정 다 했는데도 안 되는 건가요??

**Reply by 2Ho0** [reply]
아이피 문제가 있었어서 네트워크에서 호스트 어댑터로 변경했습니다
그래서 ssh-copy-id하면 연결은 되는데 이제 비번이 자꾸 오류가 뜨네요 ㅠ
엔터를 해보거나 지정한 비번 넣었는데도 안돼서 계속 찾아보는 중입니다
호스트 어댑터 말고 다른 걸로 변경 해야하는지 아니면 초기 서버 만들때 설정한 비번 말고 다른 걸로 넣어야하는지 확인중입니다

**Reply by cueon** [reply]
sudo 명령어 사용 가능한가요??

**Reply by cueon** [reply]
저도 그랬는데 `sudo passwd 현재계정명` 으로 패스워드 바꿔서 해보니까 됐어요!

**Reply by 2Ho0** [reply]
아 다행히 됐네요,, 제가 계정명을 다르게 설정했어서
`ssh-copy-id hyeong@ubuntu20-1`
`ssh-copy-id hyeonglee@ubuntu20-2`
이렇게 명시해주니까 됐습니다
계정명이 달라서 그랬네요 ㅠ
답변주셔서 감사합니다
